packages/hoprd/crates/hopr-lib/src/components/mod.rs

```diff
@@ -116,14 +117,17 @@ where
         module_implementation: Address::from_str(&chain_config.module_implementation).unwrap(),
     };
 
-    let (tx_indexer_events, rx_indexer_events) = futures::channel::mpsc::unbounded::<SignificantChainEvent>();
+    //let (tx_indexer_events, rx_indexer_events) = futures::channel::mpsc::unbounded::<SignificantChainEvent>();
+    let (mut tx_indexer_events, rx_indexer_events) = async_broadcast::broadcast(INDEXER_EVENTS_QUEUE_SIZE);
+    tx_indexer_events.set_overflow(true); // behave as ring-buffer when the capacity is reached
```

Comment:
I'm not convinced this is a good solution. It'd be better to just error out on not being able to push to the receiver. Maybe a small in-memory buffer in the indexer would be better.